lib/matplotlib/axes/_axes.py

```diff
@@ -3996,7 +3996,7 @@ def scatter(self, x, y, s=None, c=None, marker=None, cmap=None, norm=None,
             colors = None  # use cmap, norm after collection is created
 
         # c will be unchanged unless it is the same length as x:
```

Comment:
More seriously, there it pre-dates me and is almost certainly to support back compatibility.  `c` can be a scalar, a color, a sequence of colors or a sequence of scalars.  `colors` can  be (iirc) a single color or a list of colors.

`c` comes from scatter being a scalar mappable under the hood and `colors` comes from scatter returning a collection under the hood.